### PR TITLE
Add jest to reigns

### PR DESCRIPTION
**Pull request by staff0rd**
https://mobitroll.atlassian.net/browse/FRES-1145

* Adds jest
* Adds limited test coverage
* Runs in workflow

- [x] Needs #24

### CODE REVIEW

File: elements/reigns/src/features/game/gameSlice.test.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+describe("gameReducer", () => {
+  const state: GameState = {
+    ...initialState,
+    definition: {
+      cards: [{ card: "a card" } as Card],
+      stats: [{ value: 0, icon: "icon" }],
+    } as GameDefinition,
+  };
+  describe("startGame", () => {
+    it("should select next card", () => {
```
Comment (maxired) [inline elements/reigns/src/features/game/gameSlice.test.ts:13]: Looks like you created those tests only as a pretext, but probably the assertions should be more specific

File: elements/reigns/src/features/game/gameSlice.test.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      const result = gameReducer(state, startGame());
+      expect(result.selectedCard).not.toBeNull();
+    });
+  });
+  describe("answerYes", () => {
+    it("should select next card", () => {
```
Comment (maxired) [inline elements/reigns/src/features/game/gameSlice.test.ts:19]: can I answer yes before starting the game ?

File: elements/reigns/src/features/game/gameSlice.test.ts (hunk 1 above)
Comment (staff0rd) [reply elements/reigns/src/features/game/gameSlice.test.ts:13]: expect(result.selectedCard).toBe(state.definition.cards[0]);

File: elements/reigns/src/features/game/gameSlice.test.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { Card, GameDefinition, GameState } from "./types";
+
+describe("gameReducer", () => {
+  const state: GameState = {
+    ...initialState,
+    definition: {
+      cards: [{ card: "a card" } as Card],
+      stats: [{ value: 0, icon: "icon" }],
+    } as GameDefinition,
+  };
```
Comment (staff0rd) [inline elements/reigns/src/features/game/gameSlice.test.ts:11]: put this in `beforeEach`

File: elements/reigns/src/features/game/gameSlice.test.ts (hunk 2 above)
Comment (staff0rd) [reply elements/reigns/src/features/game/gameSlice.test.ts:19]: @maxired & I discussed whether each reducer should be inspecting its state before operating, similar to what a state machine would do. We could look at doing this in future PRs, but this change is only to add jest, not change functionality.

File: elements/reigns/src/features/game/gameSlice.test.ts (hunk 3 above)
Comment (staff0rd) [reply elements/reigns/src/features/game/gameSlice.test.ts:11]: moved it to `objectMother` pattern instead.

**Pull request merged**